cranelift-codegen/src/stack_layout.rs

```diff
@@ -56,6 +58,14 @@ pub fn layout_stack(frame: &mut StackSlots, alignment: StackSize) -> CodegenResu
                     .ok_or(CodegenError::ImplLimitExceeded)?;
                 outgoing_max = max(outgoing_max, offset);
             }
+            StackSlotKind::RetPtr => {
```

Comment:
This feels like a kind of silly question: where are `StackSlot` with kind `RetPtr` added? I'd expect iterating all calls in a function and adding the necessary slots from that, but don't see anything of the sort.

edit: I think I see the relevant `TODO` in legalizer/boundary.rs :)

Reply:
As a followup, and I think there ought to be a test case (or plural!) around this, if we're reusing the largest `RetPtr` slot for different multi-return calls with spills, I'd like to be sure that we know this stack slot is clobbered on reuse. That way Cranelift can spill out of this slot into somewhere that return values can be safely read from. eg `v0` should be moved out of the stack-returns area in this clif, which I think is pretty close to valid:
```
function %retptr_reuse() fast {
    sig0 = () -> i32, i32, i32, i32, i32, i32 fast
    sig1 = () -> f32, f32, f32, f32, f32, f32 fast

    ireturns = u0:0 sig0
    freturns = u0:1 sig1

ebb0():
    v0, v1, v2, v3, v4, v5 = call icalls()
    v6, v7, v8, v9, v10, v11 = call fcalls()
    ; I'm concerned v6 will overwrite v0, as things are currently
}
```

I don't know if the notion of stack slots being reused for different purposes is even something Cranelift knows about right know?

Reply:
I haven't really started on implementing the caller side yet; sorry I should have been more clear in the op.

Yeah I think it is worth not attempting to reuse these stack slots at all in the first pass implementation, and then circling back to add the reuse as a new PR.